features/100-lambda-params/api.md

```diff
@@ -0,0 +1,217 @@
+# Lambda Function with Support for Skygear Data Types
+
+## Use cases
+
+* Developer create lambda function that has Skygear Data Types
+  as parameter and return values.
+* No extra declaration is required.
+
+In this document, Skygear Data Types also include Skygear Record.
+
+## Requirements
+
+* On plugin-side, the lambda function should detect and convert JSON objects
+  into Skygear Data Types.
+* On plugin-side, the lambda function should convert Skygear Data
+  Types into standard JSON object notation (e.g. using `$type` notation).
+* In each SDK, the array/dictionary arguments should support data with Skygear
+  Data Types.
+* In each SDK, the server returned data should be converted to Skygear Data 
+  Types before calling client code.
+
+### Skygear Data Types
+
+Skygear Data Types are the following:
+
+* Skygear Record
+* Complex Data Types (example includes location, date/time)
+* Simple Data Types (example includes string, number, boolean)
+
+While not strictly a Skygear Data Type, lambda function should also support the
+following when accepting as parameter values or when returned.
+
+* array
+* dictionary
+
+### Serialization and Deserialization
+
+Serialization and Deserialization logic should share with existing
+logic implemented by record save and record fetch action. Therefore, types
+currently supported by Skygear will be supported in lambda parameters and
+return value.
+
+Since Skygear Record was not considered a data type, the protocol does
+not have support for specifying a Record object. To support using Record object
+in parameters and return value, a new `$type=record` will be implemented.
+
+### Breaking Changes
+
+When upgrading cloud function runtime (i.e. python or node), the function
+arguments will replace `$type` JSON objects with native Skygear Data Type.
+If the developer sends `$type` JSON objects from SDK, they will need to update
```

Comment:
Just want to confirm what I heard from product meeting, we are pretty sure right now most users don't do it? (Maybe a quick survey in office?)

Reply:
Did a quick survey, no one in the office will use `$type` in their lambda response, it is safe for us to do it.